src/sys/eventfd.rs

```diff
@@ -10,8 +10,58 @@ libc_bitflags! {
     }
 }
 
+#[deprecated(since = "0.27.0", note = "Use EventFd::value_and_flags() instead")]
 pub fn eventfd(initval: libc::c_uint, flags: EfdFlags) -> Result<OwnedFd> {
     let res = unsafe { libc::eventfd(initval, flags.bits()) };
 
     Errno::result(res).map(|r| unsafe { OwnedFd::from_raw_fd(r) })
 }
+
+#[derive(Debug)]
+pub struct EventFd(pub OwnedFd);
+impl EventFd {
+    /// [`EventFd::value_and_flags`] with `init_val = 0` and `flags = EfdFlags::empty()`.
+    pub fn new() -> Result<Self> {
+        let res = unsafe { libc::eventfd(0, EfdFlags::empty().bits()) };
+        Errno::result(res).map(|r| Self(unsafe { OwnedFd::from_raw_fd(r) }))
+    }
+    /// Constructs [`EventFd`] with the given `init_val` and `flags`.
+    /// 
+    /// Wrapper around [`libc::eventfd`].
+    pub fn value_and_flags(init_val: u32, flags: EfdFlags) -> Result<Self> {
+        let res = unsafe { libc::eventfd(init_val, flags.bits()) };
+        Errno::result(res).map(|r| Self(unsafe { OwnedFd::from_raw_fd(r) }))
+    }
+    /// [`EventFd::value_and_flags`] with `init_val = 0` and given `flags`.
+    pub fn flags(flags: EfdFlags) -> Result<Self> {
+        let res = unsafe { libc::eventfd(0, flags.bits()) };
+        Errno::result(res).map(|r| Self(unsafe { OwnedFd::from_raw_fd(r) }))
+    }
+    /// [`EventFd::value_and_flags`] with given `init_val` and `flags = EfdFlags::empty()`.
+    pub fn value(init_val: u32) -> Result<Self> {
+        let res = unsafe { libc::eventfd(init_val, EfdFlags::empty().bits()) };
+        Errno::result(res).map(|r| Self(unsafe { OwnedFd::from_raw_fd(r) }))
+    }
+    /// [`EventFd::write`] with `1`.
+    pub fn arm(&self) -> Result<usize> {
```

Comment:
What is the meaning of the return value, here and for defuse?

Reply:
It passes through the return value of `unistd::write`.

Reply:
Yeah, but what does that mean?  What does it even mean to write to an eventfd ?

Reply:
I think this is answered by https://github.com/nix-rust/nix/pull/1945#discussion_r1070627849